pyaedt/icepak.py

```diff
@@ -1256,130 +1256,359 @@ def create_parametric_fin_heat_sink(
         ...                                        plane_enum=icepak.PLANE.XY, rotation=45, tolerance=0.005)
 
         """
-        all_objs = self.modeler.object_names
-        self["FinPitch"] = self.modeler._arg_with_dim(pitch)
-        self["FinThickness"] = self.modeler._arg_with_dim(thick)
-        self["FinLength"] = self.modeler._arg_with_dim(length)
-        self["FinHeight"] = self.modeler._arg_with_dim(height)
-        self["DraftAngle"] = draftangle
-        self["PatternAngle"] = patternangle
-        self["FinSeparation"] = self.modeler._arg_with_dim(separation)
-        self["VerticalSeparation"] = self.modeler._arg_with_dim(vertical_separation)
-        self["HSHeight"] = self.modeler._arg_with_dim(hs_height)
-        self["HSWidth"] = self.modeler._arg_with_dim(hs_width)
-        self["HSBaseThick"] = self.modeler._arg_with_dim(hs_basethick)
+        warnings.warn(
+            "This method is deprecated in 0.7.12. Use the create_parametric_heatsink_on_face() method.",
+            DeprecationWarning,
+        )
+        rect = self.modeler.create_rectangle(plane_enum, center, [hs_width, hs_height])
+        rect.rotate(plane_enum, rotation)
+        hs, _ = self.create_parametric_heatsink_on_face(
+            rect.faces[0],
+            relative=False,
+            hs_basethick=hs_basethick,
+            fin_thick=thick,
+            fin_length=length,
+            fin_height=height,
+            draft_angle=draftangle,
+            pattern_angle=patternangle,
+            separation=separation,
+            column_separation=vertical_separation,
+            symmetric=symmetric,
+            symmetric_separation=symmetric_separation,
+            numcolumn_perside=numcolumn_perside,
+            matname=matname,
+        )
+        rect.delete()
+        return bool(hs)
+
+    @pyaedt_function_handler()
+    def create_parametric_heatsink_on_face(
+        self,
+        top_face,
+        relative=True,
+        hs_basethick=0.1,
+        fin_thick=0.05,
+        fin_length=0.25,
+        fin_height=0.5,
+        draft_angle=0,
+        pattern_angle=10,
+        separation=0.05,
+        column_separation=0.05,
+        symmetric=True,
+        symmetric_separation=0.05,
+        numcolumn_perside=2,
+        matname="Al-Extruded",
+    ):
+        """Create a parametric heat sink.
+
+        Parameters
+        ----------
+        top_face : modeler.cad.elements3d.FacePrimitive
+            Face on which to build the heatsink.
+        relative : bool, optional
+            Whether the dimensions used as arguments of the function are
+            absolute or relative to the width and the height of the
+            top_face.
+        hs_basethick : float, optional
+            Thickness of the heat sink base. If ``relative==True``, it is the
```

Comment:
why are the double equal signs required here in these statements: ``relative==True``?

Reply:
We can alternatively write
`If ``relative`` is True`

== is the comparison operator, just one = is the assignment operator

Reply:
@lorenzovecchietti I never knew this! If its technically correct, then leave it! (I hope I've never incorrectly changed it to one equal sign over the past two or three years and not have someone catch it!)